components/bms_worker/src/CAN/CANIO-tx.c

```diff
@@ -99,78 +106,68 @@ static const packTable_S MSG_packTable_1Hz[] = {
  */
 void CANTX_BUS_A_SWI(void)
 {
-    // TODO: add overrun detection here
-    if (cantx.tx_100Hz_msg == MSG_packTable_100Hz_SIZE)
+    if (cantx.tx_100Hz_msg != MSG_packTable_100Hz_SIZE)
     {
-        goto continue1;
-    }
-
-    static uint8_t counter_100Hz = 0U;
-    CAN_data_T     message_100Hz;
+        static uint8_t counter_100Hz = 0U;
+        CAN_data_T     message_100Hz;
```

Comment:
this should be initialized to zero
```suggestion
        CAN_data_T     message_100Hz = { 0U };
```